AuthorizationModelsClient is not injectable

if one tries to inject AuthorizationModelsClient, following error occurs oin startup:
```
Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkiverse.openfga.client.AuthorizationModelsClient and qualifiers [@Default]
	- injection target: io.gec.smom.permission.control.FGAUtility#authorizationModelsClient
	- declared on CLASS bean [types=[io.gec.smom.permission.control.FGAUtility, java.lang.Object], qualifiers=[@Default, @Any], target=io.gec.smom.permission.control.FGAUtility]
	at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:545)
	at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:667)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
```
As far as i can see the problem is that in OpenFGARecorder a RuntimeValue-provider-method for AuthorizationModelsClient is missing...

--- a/deployment/src/main/java/io/quarkiverse/openfga/deployment/OpenFGAProcessor.java
+++ b/deployment/src/main/java/io/quarkiverse/openfga/deployment/OpenFGAProcessor.java
@@ -7,9 +7,7 @@ import java.util.Optional;
 
 import jakarta.enterprise.context.ApplicationScoped;
 
-import io.quarkiverse.openfga.client.AuthorizationModelClient;
-import io.quarkiverse.openfga.client.OpenFGAClient;
-import io.quarkiverse.openfga.client.StoreClient;
+import io.quarkiverse.openfga.client.*;
 import io.quarkiverse.openfga.client.api.API;
 import io.quarkiverse.openfga.client.model.Store;
 import io.quarkiverse.openfga.runtime.OpenFGARecorder;
@@ -97,6 +95,20 @@ class OpenFGAProcessor {
                         .scope(ApplicationScoped.class)
                         .setRuntimeInit()
                         .runtimeValue(recorder.createAuthModelClient(apiValue, runtimeConfig))
+                        .done());
+
+        syntheticBeans.produce(
+                SyntheticBeanBuildItem.configure(AuthorizationModelsClient.class)
+                        .scope(ApplicationScoped.class)
+                        .setRuntimeInit()
+                        .runtimeValue(recorder.createAuthModelsClient(apiValue, runtimeConfig))
+                        .done());
+
+        syntheticBeans.produce(
+                SyntheticBeanBuildItem.configure(AssertionsClient.class)
+                        .scope(ApplicationScoped.class)
+                        .setRuntimeInit()
+                        .runtimeValue(recorder.createAssertionsClient(apiValue, runtimeConfig))
                         .done());
 
         return new ServiceStartBuildItem("openfga-client");

--- a/integration-tests/src/main/java/io/quarkiverse/openfga/it/OpenFGAResource.java
+++ b/integration-tests/src/main/java/io/quarkiverse/openfga/it/OpenFGAResource.java
@@ -26,10 +26,8 @@ import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.Produces;
 
-import io.quarkiverse.openfga.client.AuthorizationModelClient;
-import io.quarkiverse.openfga.client.StoreClient;
-import io.quarkiverse.openfga.client.model.AuthorizationModel;
-import io.quarkiverse.openfga.client.model.Tuple;
+import io.quarkiverse.openfga.client.*;
+import io.quarkiverse.openfga.client.model.*;
 import io.smallrye.mutiny.Uni;
 
 @Path("/openfga")
@@ -37,16 +35,39 @@ import io.smallrye.mutiny.Uni;
 public class OpenFGAResource {
 
     @Inject
+    OpenFGAClient client;
+
+    @Inject
     StoreClient storeClient;
 
     @Inject
+    AuthorizationModelsClient authorizationModelsClient;
+
+    @Inject
     AuthorizationModelClient authorizationModelClient;
+
+    @Inject
+    AssertionsClient assertionsClient;
+
+    @GET
+    @Path("stores")
+    @Produces(APPLICATION_JSON)
+    public Uni<List<Store>> listStores() {
+        return client.listAllStores();
+    }
+
+    @GET
+    @Path("changes")
+    @Produces(APPLICATION_JSON)
+    public Uni<List<TupleChange>> listChanges() {
+        return storeClient.listChanges(null, null, null);
+    }
 
     @GET
     @Path("authorization-models")
     @Produces(APPLICATION_JSON)
     public Uni<List<AuthorizationModel>> listModels() {
-        return storeClient.authorizationModels().listAll();
+        return authorizationModelsClient.listAll();
     }
 
     @GET
@@ -54,5 +75,19 @@ public class OpenFGAResource {
     @Produces(APPLICATION_JSON)
     public Uni<List<Tuple>> listTuples() {
         return authorizationModelClient.readAllTuples();
+    }
+
+    @GET
+    @Path("objects")
+    @Produces(APPLICATION_JSON)
+    public Uni<List<Tuple>> listObjects() {
+        return authorizationModelClient.readAllTuples();
+    }
+
+    @GET
+    @Path("assertions")
+    @Produces(APPLICATION_JSON)
+    public Uni<List<Assertion>> listAssertions() {
+        return assertionsClient.list();
     }
 }

--- a/integration-tests/src/test/java/io/quarkiverse/openfga/it/OpenFGAResourceTest.java
+++ b/integration-tests/src/test/java/io/quarkiverse/openfga/it/OpenFGAResourceTest.java
@@ -11,6 +11,35 @@ import io.quarkus.test.junit.QuarkusTest;
 public class OpenFGAResourceTest {
 
     @Test
+    public void testListStores() {
+
+        given()
+                .when().get("/openfga/stores")
+                .then()
+                .log().ifValidationFails()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("$", hasSize(1));
+    }
+
+    @Test
+    public void testListChanges() {
+
+        given()
+                .when().get("/openfga/changes")
+                .then()
+                .log().ifValidationFails()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("$", hasSize(3),
+                        "tuple_key.object", containsInAnyOrder("thing:1", "thing:2", "thing:2"),
+                        "tuple_key.relation", containsInAnyOrder("owner", "owner", "reader"),
+                        "tuple_key.user", containsInAnyOrder("user:me", "user:you", "user:me"),
+                        "operation",
+                        containsInAnyOrder("TUPLE_OPERATION_WRITE", "TUPLE_OPERATION_WRITE", "TUPLE_OPERATION_WRITE"));
+    }
+
+    @Test
     public void testListModels() {
 
         given()
@@ -35,5 +64,32 @@ public class OpenFGAResourceTest {
                         "key.object", containsInAnyOrder("thing:1", "thing:2", "thing:2"),
                         "key.relation", containsInAnyOrder("owner", "owner", "reader"),
                         "key.user", containsInAnyOrder("user:me", "user:you", "user:me"));
+    }
+
+    @Test
+    public void testListObjects() {
+
+        given()
+                .when().get("/openfga/objects")
+                .then()
+                .log().ifValidationFails()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("$", hasSize(3),
+                        "key.object", containsInAnyOrder("thing:1", "thing:2", "thing:2"),
+                        "key.relation", containsInAnyOrder("owner", "owner", "reader"),
+                        "key.user", containsInAnyOrder("user:me", "user:you", "user:me"));
+    }
+
+    @Test
+    public void testListAssertions() {
+
+        given()
+                .when().get("/openfga/assertions")
+                .then()
+                .log().ifValidationFails()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("$", hasSize(0));
     }
 }

--- a/runtime/src/main/java/io/quarkiverse/openfga/client/AuthorizationModelClient.java
+++ b/runtime/src/main/java/io/quarkiverse/openfga/client/AuthorizationModelClient.java
@@ -205,4 +205,8 @@ public class AuthorizationModelClient {
                 .map(WriteResponse::getValues);
     }
 
+    public AssertionsClient assertions() {
+        return new AssertionsClient(api, config);
+    }
+
 }

--- a/runtime/src/main/java/io/quarkiverse/openfga/runtime/OpenFGARecorder.java
+++ b/runtime/src/main/java/io/quarkiverse/openfga/runtime/OpenFGARecorder.java
@@ -4,10 +4,7 @@ import static io.quarkiverse.openfga.client.OpenFGAClient.storeIdResolver;
 
 import java.util.function.Supplier;
 
-import io.quarkiverse.openfga.client.AuthorizationModelClient;
-import io.quarkiverse.openfga.client.ClientConfig;
-import io.quarkiverse.openfga.client.OpenFGAClient;
-import io.quarkiverse.openfga.client.StoreClient;
+import io.quarkiverse.openfga.client.*;
 import io.quarkiverse.openfga.client.api.API;
 import io.quarkiverse.openfga.runtime.config.OpenFGAConfig;
 import io.quarkus.runtime.RuntimeValue;
@@ -44,5 +41,19 @@ public class OpenFGARecorder {
                         .map(modelId -> new ClientConfig(storeId, modelId)));
         var authModelClient = new AuthorizationModelClient(api.getValue(), configResolver);
         return new RuntimeValue<>(authModelClient);
+    }
+
+    public RuntimeValue<AuthorizationModelsClient> createAuthModelsClient(RuntimeValue<API> api, OpenFGAConfig config) {
+        var storeIdResolver = storeIdResolver(api.getValue(), config.store(), config.alwaysResolveStoreId());
+        var authModelsClient = new AuthorizationModelsClient(api.getValue(), storeIdResolver);
+        return new RuntimeValue<>(authModelsClient);
+    }
+
+    public RuntimeValue<AssertionsClient> createAssertionsClient(RuntimeValue<API> api, OpenFGAConfig config) {
+        var configResolver = storeIdResolver(api.getValue(), config.store(), config.alwaysResolveStoreId())
+                .flatMap(storeId -> OpenFGAClient.authorizationModelIdResolver(api.getValue(), storeId)
+                        .map(modelId -> new ClientConfig(storeId, modelId)));
+        var assertionsClient = new AssertionsClient(api.getValue(), configResolver);
+        return new RuntimeValue<>(assertionsClient);
     }
 }

